### PR TITLE
chore: added oci label to Dockerfile to reference the source

### PR DESCRIPTION
**Pull request by kvanzuijlen**
This label could be used by (for example) [Renovatebot](https://docs.renovatebot.com/modules/datasource/docker/) to automatically pull a changelog into PRs it creates.

### CODE REVIEW

**Review by kaovilai — COMMENTED** [review]
can you also dynamically add this?
```
org.opencontainers.image.created
```

This is static
```
org.opencontainers.image.documentation="https://velero.io"
```

**Comment by kvanzuijlen** [comment]
> can you also dynamically add this?
> 
> ```
> org.opencontainers.image.created
> ```
> 
> This is static
> 
> ```
> org.opencontainers.image.documentation="https://velero.io"
> ```

Sure! I can also add other oci annotations if desired

This are all the oci annotations that could apply (I think):
-     org.opencontainers.image.created (current date)
-     org.opencontainers.image.authors (probably a static list?)
-     org.opencontainers.image.url (either gcr.io/velero-gcp/velero-plugin-for-gcp, hub.docker.com/r/velero/velero-plugin-for-gcp or velero.io)
-     org.opencontainers.image.documentation (https://velero.io)
-     org.opencontainers.image.version (same as version-tag for image (value from docker_push.sh))
-     org.opencontainers.image.revision (git short sha)
-     org.opencontainers.image.vendor (vmware-tanzu)
-     org.opencontainers.image.licenses (Apache-2.0)
-     ~org.opencontainers.image.ref.name~
-     org.opencontainers.image.title (velero-plugin-for-gcp)
-     org.opencontainers.image.description (Plugins to support Velero on Google Cloud Platform (GCP))
-     ~org.opencontainers.image.base.digest~
-     org.opencontainers.image.base.name (scratch)

**Review by blackpiglet — DISMISSED** [review]

**Comment by kvanzuijlen** [comment]
@kaovilai @blackpiglet shall I open a separate PR for additional labels? Which ones shall I add?

**Comment by blackpiglet** [comment]
If more labels are needed, I prefer to put them in the same PR.
I don't know what information is needed. @kaovilai Please share some insight.

**Comment by kvanzuijlen** [comment]
> I don't think there's a need. Just nice to have. Same pr is fine.

Which of the above listed labels? Only the ones you requested or some additional ones?

**Comment by kaovilai** [comment]
I have no opinion on this other than that it should be labels that do not require manual update every release. If all we have now is all that's desired, we can stick with that for now, if there are others that are relatively low effort, I wouldn't mind it being added to this PR.

**Comment by kvanzuijlen** [comment]
I ran the build locally, this is what the resulting labels look like with docker inspect:
image
created date/time is UTC.

**Review by kaovilai — APPROVED** [review]
lgtm. For consistency probably add to other repos if this merges.

File: Dockerfile [hunk 1]
```diff
@@ -35,3 +35,20 @@ COPY --from=build /go/bin/velero-plugin-for-gcp /plugins/
...
 ENTRYPOINT ["cp-plugin", "/plugins/velero-plugin-for-gcp", "/target/velero-plugin-for-gcp"]
+
+ARG CREATED
+ARG VERSION
+ARG GIT_SHA
+
+LABEL \
+	org.opencontainers.image.created=${CREATED} \
+	org.opencontainers.image.url="https://hub.docker.com/r/velero/velero-plugin-for-gcp" \
+	org.opencontainers.image.documentation="https://velero.io" \
```
Comment (kaovilai) [inline Dockerfile:46]: Actually, for this image specifically, doc should be `https://github.com/vmware-tanzu/velero-plugin-for-gcp` imo.

**Review by kaovilai — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**